### PR TITLE
Remove PA dependency on OpenSearch Process and add DISABLE PA parameter on Docker

### PR DESCRIPTION
**Pull request by peterzhuamazon**
Signed-off-by: Peter Zhu <zhujiaxi@amazon.com>

### Description
 Remove PA dependency on OpenSearch Process and add DISABLE PA parameter on Docker

### Issues Resolved
https://github.com/opensearch-project/opensearch-build/issues/1185

By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.
For more information on following Developer Certificate of Origin and signing off your commits, please check [here](https://github.com/opensearch-project/OpenSearch/blob/main/CONTRIBUTING.md#developer-certificate-of-origin).

### CODE REVIEW

**Comment by dblock** [comment]
There seems to be a bunch of other changes here with removal of `terminateProcesses`.

**Comment by peterzhuamazon** [comment]
> There seems to be a bunch of other changes here with removal of `terminateProcesses`.

Hi @dblock
1. Remove PA OS coupling
2. Add new param for user to disable PA Agent Cli upon start
3. Add Readme.

File: docker/release/config/opensearch/opensearch-docker-entrypoint.sh [hunk 1]
```diff
@@ -49,21 +45,23 @@ function setupSecurityPlugin {
...
+    if [ -d "$OPENSEARCH_HOME/plugins/$PERFORMANCE_ANALYZER_PLUGIN" ]; then
+        if [ "$DISABLE_PERFORMANCE_ANALYZER_AGENT_CLI" = "true" ]; then
+            echo "Disabling execution of $OPENSEARCH_HOME/bin/$PERFORMANCE_ANALYZER_PLUGIN/performance-analyzer-agent-cli for OpenSearch Performance Analyzer Plugin"
+        else
+            echo "Enabling execution of OPENSEARCH_HOME/bin/$PERFORMANCE_ANALYZER_PLUGIN/performance-analyzer-agent-cli for OpenSearch Performance Analyzer Plugin"
+            $OPENSEARCH_HOME/bin/opensearch-performance-analyzer/performance-analyzer-agent-cli > $OPENSEARCH_HOME/logs/performance-analyzer.log 2>&1 &
+        fi
+    else
+        echo "OpenSearch Performance Analyzer Plugin does not exist, disable by default"
     fi
```
Comment (sgup432) [inline docker/release/config/opensearch/opensearch-docker-entrypoint.sh:65]: Shouldn't we retain this logic to terminate in case signal is received?

Comment (peterzhuamazon) [reply docker/release/config/opensearch/opensearch-docker-entrypoint.sh:65]: The whole point of this PR is, if PA terminated do not even change the behavior of OS.
OS can keep running while PA terminated.

The initial PA/OS coupled logic is out of date now, it was meant to replicate supervisord behavior previously.

**Review by rishabh6788 — APPROVED** [review]

**Pull request merged**